File: RedSheet.DbEntities/Models/vProjectEventTimelineRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vProjectEventTimelineRecords", Schema = "dbo")]
	[Serializable]
	public partial class vProjectEventTimelineRecord  
	{
 
	[Required]
	public TimeSpan EndTime { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string EventDuration { get; set; } 
 
	[Required]
	public bool IsEndTimeFixed { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string OpeningStatement { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ProjectEventTimelineId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string RoomLayout { get; set; } 
 
	[Required]
	public TimeSpan StartTime { get; set; } 

	public vProjectEventTimelineRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-project-event-timeline-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vProjectEventTimelineRecord {
    constructor(vProjectEventTimelineRecord?: vProjectEventTimelineRecord )  {
        let properties = [ "endTime", "eventDuration", "isEndTimeFixed", "openingStatement", "projectEventTimelineId", "projectModuleId", "roomLayout", "startTime",];
        for (let property of properties)
            if (vProjectEventTimelineRecord && vProjectEventTimelineRecord[property])
                this[property] = vProjectEventTimelineRecord[property];
    }
 
    @required()
	endTime : Date =   undefined;
 
    @required()
    @maxLength(200)
	eventDuration : string =   undefined;
 
	isEndTimeFixed : boolean = false ;
 
    @required()
    @maxLength(1000)
	openingStatement : string =   undefined;
 
	projectEventTimelineId : number =   0 ;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;
 
    @required()
    @maxLength(1000)
	roomLayout : string =   undefined;
 
    @required()
	startTime : Date =   undefined;


}
